src/Microsoft.DotNet.Build.Tasks.Packaging/src/VersionUtility.cs

```diff
@@ -41,5 +49,25 @@ public static Version GetAssemblyVersion(string assemblyPath)
 
             return result;
         }
+
+        public static Version As3PartVersion(Version version)
+        {
+            if (version.Revision == -1)
+            {
+                return version;
+            }
+
+            return new Version(version.Major, version.Minor, version.Build);
```

Comment:
It's not all that interesting for our scenarios, but I'll add safegaurds since this is a utility method.